File: Site/Gmf.Marush.Care.Api/Models/Calendar/CalendarEntryRequest.cs
namespace Gmf.Marush.Care.Api.Models.Calendar;
public record CalendarEntryRequest
{
    public DateOnly Date { get; init; }
    public TimeOnly StartTime { get; init; }
    public TimeOnly EndTime { get; init; }
    public Guid? CustomerId { get; init; }
    public Guid? AppointmentId { get; init; }
    public string? Notes { get; init; }
    public decimal? Money { get; init; }
}

File: Site/Gmf.Marush.Care.App/src/app/features/admin/calendar/calendar-entry-request.ts
import { EntryFormValue } from './entry-dialog/entry-form-value';

export class CalendarEntryRequest {
    date: string = '';
    startTime: string = '';
    endTime: string = '';
    customerId?: string;
    appointmentId?: string;
    notes?: string;
    money?: number;
    treatments: string[] = [];

    static from = (formValue: EntryFormValue, treatments: string[]): CalendarEntryRequest => Object.assign(new CalendarEntryRequest(), {
            date: formValue.date,
            startTime: formValue.startTime,
            endTime: formValue.endTime,
            customerId: formValue.customerId || undefined,
            appointmentId: formValue.appointmentId || undefined,
            notes: formValue.notes || undefined,
            money: formValue.money ?? undefined,
            treatments
        });
}
